client-src/overlay/styles.js

```diff
@@ -24,58 +16,131 @@ const iframeStyle = {
+      overflow: auto;
+      background-color: rgba(0, 0, 0, 0.9);
+      color: white;
+    }
+  `,
 };
 
 const headerStyle = {
-  color: "#e83b46",
-  fontSize: "2em",
-  whiteSpace: "pre-wrap",
-  fontFamily: "sans-serif",
-  margin: "0 2rem 2rem 0",
-  flex: "0 0 auto",
-  maxHeight: "50%",
-  overflow: "auto",
+  className: "webpack-header",
+  css: /* css */ `.webpack-header {
+      color: #e83b46;
+      font-size: 2em;
+      font-family: sans-serif;
+      white-space: pre-wrap;
+      margin: 0 2rem 2rem 0;
+      flex: 0 0 auto;
+      max-height: 50%;
+      overflow: auto;
+    }
+  `,
 };
 
 const dismissButtonStyle = {
-  color: "#ffffff",
-  lineHeight: "1rem",
-  fontSize: "1.5rem",
-  padding: "1rem",
-  cursor: "pointer",
-  position: "absolute",
-  right: 0,
-  top: 0,
-  backgroundColor: "transparent",
-  border: "none",
+  className: "webpack-dismiss-btn",
+  css: /* css */ `.webpack-dismiss-btn {
+    color: #ffffff;
+    line-height: 1rem;
+    font-size: 1.5rem;
+    padding: 1rem;
+    cursor: pointer;
+    position: absolute;
+    right: 0;
+    top: 0;
+    background-color: transparent;
+    border: none;
+  }
+  
+  .webpack-dismiss-btn:hover {
+    color: #d1d5db;
+  }`,
 };
 
 const msgTypeStyle = {
-  color: "#e83b46",
-  fontSize: "1.2em",
-  marginBottom: "1rem",
-  fontFamily: "sans-serif",
+  className: "webpack-msg-type",
+  css: /* css */ `.webpack-msg-type {
+      margin-bottom: 1rem;
+      color: #e83b46;
+      font-size: 1.2em;
+      font-family: sans-serif;
+    }
+
+  .webpack-msg-type[data-can-open] {
+    cursor: pointer;
+  }
+  `,
 };
 
 const msgTextStyle = {
-  lineHeight: "1.5",
-  fontSize: "1rem",
-  fontFamily: "Menlo, Consolas, monospace",
+  className: "webpack-msg-text",
+  css: /* css */ `.webpack-msg-text {
+      line-height: 1.5;
+      font-size: 1rem;
+      font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
+    }
+  `,
+};
+
+const msgStyles = {
+  error: {
+    className: "webpack-error-msg",
+    css: /* css */ `.webpack-error-msg {
+      background-color: rgba(206, 17, 38, 0.1);
+      color: #fccfcf;
+      padding: 1rem 1rem 1.5rem 1rem;
+    }`,
+  },
+  warning: {
+    className: "webpack-warning-msg",
+    css: /* css */ `.webpack-warning-msg {
+      background-color: rgba(251, 245, 180, 0.1);
+      color: #fbf5b4;
+      padding: 1rem 1rem 1.5rem 1rem;
+    }`,
+  },
+};
+
+/**
+ * @typedef {Object} CssLoader
+ * @property {(css: string) => void} load
+ */
+
+/**
+ *
+ * @param {Document} doc
+ * @return {CssLoader}
+ */
+const createCssLoader = (doc) => {
+  /** @type {string[]} */
+  const loadedCss = [];
+
+  return {
+    load: (css) => {
+      // ignore CSS rule that has loaded before
+      if (!loadedCss.includes(css)) {
+        const style = doc.createElement("style");
+        style.innerHTML = css;
+        doc.head.appendChild(style);
+        loadedCss.push(css);
+      }
+    },
+  };
```

Comment:
It can create a lot of problems. Shorty - we already tried this solution and we had a lot of issues about `style` tags and its order and overlapping and etc and a lot, so I prefer do not use style and classes at all, I am fine with refactor and move styles in own objects with good names, but using classes again make more problems than benefits

Reply:
Got it. Any issues that was raised before on this? Maybe worth include them in the comments

Reply:
They were, but they are difficult to find because they are scattered and many headings do not match the original problem, I tried to search for them, but unfortunately I can not find but I remember very well that we had problems with this :smile: